Cocoaheads School_Lesson2.playground/Contents.swift

```diff
@@ -19,31 +19,60 @@ class Stack {
 }
 
 class StackStatistics: Stack {
-    private var minValues = [Int]()
-
+    private var minElement: Int?
+    
     override func push(element: Int) {
-        super.push(element: element)
-        if let minValue = minValues.last {
+        if let minValue = minElement {
             if minValue > element {
-                minValues.append(element)
+                super.push(element: (2*element - minValue))
+                minElement = element
             } else {
-                minValues.append(minValue)
+                super.push(element: element)
             }
         } else {
-            minValues.append(element)
+            minElement = element
+            super.push(element: element)
         }
     }
     
     override func pop() -> Int? {
-        guard !minValues.isEmpty else {
-            return nil
+        if let elementTop = super.top() {
+            if let minValue = minElement {
```

Comment:
```suggestion
            if let minElement = minElement {
```
Do not introduce new entities here. It's harder to track them all.

Reply:
done